tests/phpunit/tests/image/editorImagick.php

```diff
@@ -641,4 +641,53 @@ public function test_directory_creation() {
 
 		$this->assertNotWPError( $saved );
 	}
+
+	/**
+	 * Tests that the alpha channel of PDFs is removed from PDF previews.
+	 *
+	 * Only affects systems with Ghostscript version >= 9.14.
+	 *
+	 * @ticket 39216
+	 *
+	 * @covers WP_Image_Editor_Imagick::remove_pdf_alpha_channel
+	 */
+	public function test_remove_pdf_alpha_channel_should_remove_the_alpha_chanel_in_preview() {
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'application/pdf' ) ) ) {
+			$this->markTestSkipped( 'Rendering PDFs is not supported on this system.' );
+		}
+
+		$test_file     = DIR_TESTDATA . '/images/test-alpha.pdf';
+		$attachment_id = $this->factory->attachment->create_upload_object( $test_file );
+		$this->assertNotEmpty( $attachment_id, 'The attachment was not created before testing.' );
+
+		$attached_file = get_attached_file( $attachment_id );
+		$this->assertNotEmpty( $attached_file, 'The attached file was not returned.' );
+
+		$rgb = array(
+			'r' => true,
+			'g' => true,
+			'b' => true,
+		);
+
+		// White.
+		$expected = array(
+			'r' => 1,
+			'g' => 1,
+			'b' => 1,
+		);
+
+		$check = image_get_intermediate_size( $attachment_id, 'full' );
+		$this->assertNotEmpty( $check['file'], 'The intermediate size file was not found.' );
```

Comment:
Unexpectedly, when running this test locally (WSL2, Ubuntu), I'm getting this:

```php
1) Tests_Image_Editor_Imagick::test_remove_pdf_alpha_channel_should_remove_the_alpha_chanel_in_preview
Trying to access array offset on value of type bool
```
It seems that `wp_get_attachment_metadata()` is returning `array( 'filesize' => 7175 )` for me locally, thereby missing the `'sizes'` key, and causing `wp_get_attachment_metadata()` to return `false`. I'm not sure why that is. Do you have any ideas why that could happen?

We might need to look at how this can happen, and see if something needs to be stabilised.

Additionally, we might want to change this to:
```php
$this->assertIsArray( $check, 'The intermediate size could not be retrieved.' )
$this->assertArrayHasKey( 'file', $check, 'The intermediate size file was not found.' );
```

Even though this would fail immediately for me locally due to the above issue, it will at least catch it without the test erroring out.